tests/baselines/reference/privateNameFieldCallExpression.js

```diff
@@ -0,0 +1,33 @@
+//// [privateNameFieldCallExpression.ts]
+class A {
+    #fieldFunc = function() { this.x = 10; };
+    x = 1;
+    test() {
+        this.#fieldFunc();
+        const func = this.#fieldFunc;
+        func();
+    }
+}
+
+
+//// [privateNameFieldCallExpression.js]
+var __classPrivateFieldGet = (this && this.__classPrivateFieldGet) || function (receiver, privateMap) {
+    if (!privateMap.has(receiver)) {
+        throw new TypeError("attempted to get private field on non-instance");
+    }
+    return privateMap.get(receiver);
+};
+var _fieldFunc;
+var A = /** @class */ (function () {
+    function A() {
+        _fieldFunc.set(this, function () { this.x = 10; });
+        this.x = 1;
+    }
+    A.prototype.test = function () {
+        __classPrivateFieldGet(this, _fieldFunc).call(this);

```

Comment:
I'd like to see the following additional test cases:

- argument spread in call (i.e. `this.#field(a, ...args, b)`)
- `new` (i.e. `new this.#field()`)
- argument spread in `new` (i.e. `new this.#field(a, ...args, b)`)
- tagged template expressions (i.e. `` this.#field`head${1}middle${2}tail` ``)

Reply:
👍 thank you! This pointed out an issue with the `this` binding of tagged template expressions (where the tag is a private identifier property access). I fixed with [this commit](https://github.com/microsoft/TypeScript/pull/30829/commits/2e539b98490861afc1a928ca32ffc4680ac36a12).